docker-compose.yml

```diff
@@ -3,50 +3,67 @@
 #
 version: '2.1'
 
+# run with:
+# > docker-compose run --service-ports --rm --name node_main node_main /bin/bash
```

Comment:
Is there a way to run this without starting main or waiting on it?

what does `--rm` do?

also, i've never used `docker-compose run` just `docker-compose up -d` or `docker-compose down`. is it pretty much the same as `docker run -it node_main /bin/bash`?

Reply:
`docker-compose up -d` works too, it starts the main container with the command defined in the Dockerfile. 
I like to have this command handy because it gets me into a shell.
It is like `run -it` but starts up the container, the network, and all the container it depends on. When you ctrl-d it doesn't shut everything down, but --rm removes this container and this command can be used to start it and join the network again. It may work without --rm, but it is not documented that way.
I like this way of starting up because instead of /bin/bash it can also be a script. I'm working on running `. env.sh docker` on startup. In ruby I have a script that will set up everything and depending in the argument will run all tests or start a shell.